### ISSUE TITLE
RTSP buffer

### ISSUE THREAD

**Issue by jgoo9410** [ultralytics/ultralytics#4211]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Detection

### Bug

Somewhere between version 8.0.55 and the current version, the behaviour of the RTSP functionality as a stream source has changed. 

Previously, using an RTSP stream for realtime detection would grab the 'current' or most recent frame from the stream, but in the latest version, (could have changed any version between 8.0.55 and the current), the functionality now appears to buffer the frames instead of just picking the most recent. The result is that, if a single iteration of detection takes longer than the time between consecutive frames, a frame buffer begins to fill, and the video slows down. Eventually the buffer becomes full, the video is several minutes behind, and the interpreter crashes. 

This makes using the built in RTSP detection functionality effectively useless for any device that isn't capable of processing a single iteration of the detection process at a rate that is consistent with the frame rate of the stream. 

Was this a conscious change, or a version change from a dependent library, such as open-CV?

`for result in model.track( device
"ap:
source=stream_ source, stream = True, conf=0.4, 1ou =0, tracker= current_directory+'/tracker_config.yam', show false,verbose = False):`

### Environment

Ultralytics - 8.0.55 - current
Python - 3.10.12
Mac OS (apple silicon)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@jgoo9410 probably https://github.com/ultralytics/ultralytics/pull/3602 by myself.

Maybe we should add an argument that controls this behavior, i.e. `buffer_streams=True` or `buffer=True` or `stream=buffer, stream=latest`?

**Comment by jgoo9410** [comment]
@glenn-jocher Ah, that explains it.

That would be very helpful. `stream_buffer=True/False` would probably suffice? 

As far as I'm aware, RTSP streaming is predominantly used for CCTV camera access, so having a buffer is unlikely to be useful to most. In fact, for any live streaming it will invariably cause and issue. 

What was the change made for?

**Comment by glenn-jocher** [comment]
@jgoo9410 yeah that's true. Other non-live streaming sources like YouTube videos were skipping frames before, so PR #3602 resolved this, i.e. see issue #https://github.com/ultralytics/ultralytics/issues/3559.

Ok I'll add a TODO here. If you'd like to help with a PR that would be super helpful as I'm super busy currently, but if not I should be able to get to this at some point in the future.

**Comment by jgoo9410** [comment]
@glenn-jocher I will certainly have a look, I'll let you know how I get on.

**Comment by glenn-jocher** [comment]
@jgoo9410 Thank you for your willingness to contribute to the project. Your efforts in addressing this issue are certainly appreciated. If you encounter any issues or have further questions during your journey, don't hesitate to ask. Looking forward to seeing your progress. Stay in touch!

**Comment by jgoo9410** [comment]
@glenn-jocher sorry for the delay. Done now. Functionality as described above: optional arg 'stream_buffer' (bool), default False. 
I havent touched any docs, not  sure where you want this feature documented.

Tested with:
```
for result in model.track(device = "mps",source='rtsp://admin:orchard123@192.168.0.50:554/Streaming/Channels/101/',stream = True,conf=0.5, iou =0, show=True,verbose = False, stream_buffer = False):
        time.sleep(0.5)
```

**Comment by glenn-jocher** [comment]
@jgoo9410 awesome! PR is now merged to main and published in `ultralytics 8.0.164`. Nice work!! :)
